docs/source/gateway/index.rst

```diff
@@ -1,6 +1,782 @@
+
+More providers are being added continually. Check the latest version of the MLflow AI Gateway Docs for the
+most up-to-date list of supported providers.
+
+Remember, the provider you specify must be one that the MLflow AI Gateway supports. If the provider
+is not supported, the Gateway will return an error when trying to route requests to that provider.
+
+Routes
+------
+
+`Routes` are central to how the MLflow AI Gateway functions. Each route acts as a proxy endpoint for the
+user, forwarding requests to the underlying `model` and `provider` specified in the configuration file.
+
+A route in the MLflow AI Gateway consists of the following fields:
+
+* **name**: This is the unique identifier for the route. This will be part of the URL when making API calls via the MLflow AI Gateway.
+
+* **type**: The type of the route corresponds to the type of language model interaction you desire. For instance, ``llm/v1/completions`` for text completion operations, ``llm/v1/embeddings`` for text embeddings, and ``llm/v1/chat`` for chat operations.
+
+* **model**: Defines the model to which this route will forward requests. The model contains the following details:
+
+    * **provider**: Specifies the name of the :ref:`provider <providers>` for this model. For example, ``openai`` for `OpenAI`'s ``GPT-3`` models.
+    * **name**: The name of the model to use. For example, ``gpt-3.5-turbo`` for `OpenAI`'s ``GPT-3.5-Turbo`` model.
+    * **config**: Contains any additional configuration details required for the model. This includes specifying the API base URL and the API key.
+
+Here's an example of a route configuration:
+
+.. code-block:: yaml
+
+    routes:
+        - name: completions
+          type: chat/completions
+          model:
+            provider: openai
+            name: gpt-3.5-turbo
+            config:
+              openai_api_base: https://api.openai.com/v1
+              openai_api_key: $OPENAI_API_KEY
+
+In the example above, a request sent to the completions route would be forwarded to the
+``gpt-3.5-turbo`` model provided by ``openai``.
+
+The routes in the configuration file can be updated at any time, and the MLflow AI Gateway will
+automatically update its available routes without requiring a restart. This feature provides you
+with the flexibility to add, remove, or modify routes as your needs change. It enables 'hot-swapping'
+of routes, providing a seamless experience for any applications or services that interact with the MLflow AI Gateway.
+
+When defining routes in the configuration file, ensure that each name is unique to prevent conflicts.
+Duplicate route names will raise an ``MlflowException``.
+
+Models
+------
+
+The ``model`` section within a ``route`` specifies which model is to be used for generating responses.
+This configuration block needs to contain a ``name`` field which is used to specify the exact model instance to be used.
```

Comment:
It also needs to contain a ``provider`` field. Can we mention that briefly here and link to the above "Providers" section?